### ISSUE TITLE
AOCC compiler cannot build HDF5

### ISSUE THREAD

**Issue by lcebaman** [HDFGroup/hdf5#3439]
**Describe the bug**
AOCC compiler is not able to build HDF5

**Platform (please complete the following information)**
-HDF5 develop
-AOCC 4.0
-OpenMPI 4.1.5

**How was configured**

```
export FC=mpif77
export F90=mpif90
export F77=mpif77
export CXX=mpic++
export LD=mpif90
export MPIF90=mpif90
export OMPI_CC=clang
export OMPI_FC=flang
export OMPI_CXX=clang++
./configure  --enable-parallel --enable-fortran --enable-shared --enable-static
```

**Error**
```
clang: error: no such file or directory: '../src/.libs/libhdf5.so'
make[1]: *** [Makefile:1979: cache] Error 1
make[1]: *** Waiting for unfinished jobs....
clang: error: no such file or directory: '../src/.libs/libhdf5.so'
clang: error: no such file or directory: '../src/.libs/libhdf5.so'
make[1]: *** [Makefile:1987: cache_image] Error 1
make[1]: *** [Makefile:1983: cache_api] Error 1
clang: error: no such file or directory: '../src/.libs/libhdf5.so'
make[1]: *** [Makefile:1995: cache_tagging] Error 1
clang: error: no such file or directory: '../src/.libs/libhdf5.so'
clang: error: no such file or directory: '../src/.libs/libhdf5.so'
make[1]: *** [Makefile:2239: lheap] Error 1
make[1]: *** [Makefile:2283: ohdr] Error 1
clang: error: no such file or directory: '../src/.libs/libhdf5.so'
make[1]: *** [Makefile:2319: stab] Error 1
```

**Comment by hyoklee** [comment]
Hi, @lcebaman !

Thank you for testing AOCC with HDF5.
I could confirm the build failure.

However, if you disable shared  & Fortran & parallel, you can build and test HDF5 with the latest AOCC.
Here's an example GitHub Action result that passes:

https://github.com/hyoklee/actions/actions/runs/6014286859/job/16313640693 

Here's my test Action workflow for the above result: 
https://github.com/hyoklee/actions/blob/main/.github/workflows/aocc.yml

**Comment by lcebaman** [comment]
Hi @hyoklee 

Just to confirm, you were able to build HDF5 with AOCC but with Fortran, parallel and shared options disabled.

**Comment by hyoklee** [comment]
@lcebaman Yes.

**Comment by derobins** [comment]
This is/was a libtool error. The libtool script that is generated at compile time was missing the "-Wl," prefix necessary for linker options, so "-Wl,-soname" gets parsed as -soname, which fails.

As of October 2023, with AOCC 4.1, c/flang works out of the box w/ shared unless you are using an MPI compiler wrapper. We have a hack for that (at least in the Autotools), which will appear shortly.

This issue will be closed when the PR is merged to develop.

**Comment by derobins** [comment]
Fixed w/ #3674
